inginious/frontend/plugins/task_hints/pages/api/hints_mode_api.py

```diff
@@ -0,0 +1,67 @@
+import web
+
+from inginious.frontend.pages.api._api_page import APIError
+from inginious.frontend.plugins.utils.admin_api import AdminApi
+from inginious.frontend.plugins.utils import get_mandatory_parameter
+from ..user_hint_manager import UserHintManagerSingleton
+
+class HintsModeAPI(AdminApi):
+
+    @property
+    def user_hint_manager(self) -> UserHintManagerSingleton:
+        """ Returns user hint manager singleton """
+        return UserHintManagerSingleton.get_instance()
+
+    def API_GET(self):
+        
+        """
+            Get task submission mode before it's saved
+        """
+
+        input_data = web.input()
+
+        course_id = get_mandatory_parameter(input_data, 'course_id')
+        task_id = get_mandatory_parameter(input_data, 'task_id')
+
+        course = self.get_course_and_check_rights(course_id)
+
+        try:
+            task = self.task_factory.get_task(course, task_id)
+            task_submission_mode = task._data.get('groups')
```

Comment:
```suggestion
            task_submission_mode = task._data.get('groups', False)
```
Do this for the other cases where you are accessing "groups"